pkg/provider/certificate_authority.go

```diff
@@ -206,22 +199,32 @@ func NewCA(ctx context.Context, opts NewCAOptions) (*FullCertificateAuthority, e
 
 // Save saves a CA with the given configuration
 func (fc FullCAConfig) Save(ca *FullCertificateAuthority) error {
-	var certData, keyData bytes.Buffer
+	var (
+		certData, keyData                                              bytes.Buffer
+		writeChainErr, writeChainDataErr, writeKeyErr, writeKeyDataErr error
+	)
 
 	chain := []*x509.Certificate{ca.Cert}
 	chain = append(chain, ca.RestChain...)
 
-	writeErr := utils.CombineErrors(
-		peertls.WriteChain(&certData, chain...),
-		peertls.WriteKey(&keyData, ca.Key),
-	)
+	if fc.CertPath != "" {
+		writeChainErr = peertls.WriteChain(&certData, chain...)
+		writeChainDataErr = writeChainData(fc.CertPath, certData.Bytes())
```

Comment:
What if `writeChainErr != nil`? What's the point of attempting `writeChainData` in this case?